lib/p2p/Pool.ts

```diff
@@ -371,14 +395,27 @@ class Pool extends EventEmitter {
       peer.sendPacket(new packets.GetOrdersPacket());
       peer.sendPacket(new packets.GetNodesPacket());
 
+      // updating the `lastConnected` field for the address we're actually connected to.
+      // if connection is inbound, then the port is discarded. This might result in multiple addresses getting updated
+      const addresses = peer.addresses!.map((address) => {
+        if (peer.socketAddress.host === address.host &&
+            peer.inbound ? true : peer.socketAddress.port === address.port
```

Comment:
I don't think we should be updating `lastConnected` on inbound connections, as you point out it might update multiple addresses, but mainly the inbound connection doesn't actually verify that we're able to connect to their external address, which is what I'm thinking `lastConnected` should track

Reply:
Makes sense, i'll change it